src/reducers/dependencies.reducer.js

```diff
@@ -45,6 +45,7 @@ export default (state: State = initialState, action: Action) => {
           // All of the other fields are unknown at this point.
           // To make life simpler, we'll set them to empty strings,
           // rather than deal with nullable fields everywhere else.
+          location: 'dependencies',
           description: '',
           version: '',
           homepage: '',
```

Comment:
Ahhh, ok, so the value being set here is used in the 'interim', but then `ADD_DEPENDENCY_FINISH` below - the action that sets the "permanent" value - just uses the dependency value from the action (line 72).

Tracing it back, in `dependency.middleware.js` line 30, I see:

```js
        .then(() => loadProjectDependency(projectPath, dependencyName))
```

No `dependencyLocation` is provided here, and so that's why the value is ending up as `undefined`.

Two possible fixes:

- Update `dependency.middleware.js` line 30 to provide the default value:

```js
        .then(() => loadProjectDependency(projectPath, dependencyName, 'dependencies'))

```

- Set a default value for `loadProjectDependency` in `read-from-disk.service.js`:

```js
export function loadProjectDependency(
  projectPath: string,
  dependencyName: string,
  dependencyLocation: DependencyLocation = 'dependencies'
) {
```

I don't have a strong preference between the two. I think I have a minor preference for the latter option, the default value, since we'll want to use that value in _almost_ every case, and it'll make it easier to use `loadProjectDependency` elsewhere. But yeah, I'm happy either way :)

Reply:
Went with option 2. Looks like this took care of the problem I originally raised with the default value not being set, so I also removed my prior code in the component file that attempted to deal with it 🙂